python/mxnet/executor_manager.py

```diff
@@ -283,8 +283,13 @@ def update_metric(self, metric, labels):
         """ Update evaluation metric with label and current outputs """
         for texec, islice in zip(self.train_execs, self.slices):
             n = int(texec.outputs[0].shape[0] / (islice.stop - islice.start))
-            new_slice = slice(islice.start * n, islice.stop * n)
-            labels_slice = [label[new_slice] for label in labels]
+            # special case for ctc loss
+            if len(labels) > 0 and len(labels[0].shape) == 2 \
+               and n > labels[0].shape[1]:
```

Comment:
I lstm+ctc case, n is not batch_size. n is sequence length. labels[0]'s shape is (batch_size, label_size).

So, @precedenceguo 's fix is not work for lstm+ctc case.

Without CTC, sequence length should equal to label size. So, my fix means, with CTC, sequence length > label size.

However, the best fix will be:

if self.is_ctc:
then do something.

But I do not know how to add is_ctc property into self, do you have any ideas?